src/cache.rs

```diff
@@ -14,14 +15,22 @@ lazy_static! {
     static ref CACHE: RwLock<HashMap<Generator, CacheItem>> = RwLock::new(HashMap::new());
 }
 
-pub fn get<T>(generator: Generator) -> Result<T, Box<dyn Error>>
+pub async fn get<T>(generator: Generator) -> Result<T, Box<dyn Error>>
 where
     T: Send + Sync + Clone + 'static,
 {
     if let Some(cached) = get_cached(generator) {
         Ok(cached)
     } else {
-        update_cache(generator)
+        task::spawn_blocking(move || {
+            update_cache::<T>(generator)
+                // stringify the error to make it Send
+                .map_err(|e| e.to_string())
+        })
+        .await
+        .map_err(Box::new)?
+        // put the previously stringified error back in a box
+        .map_err(|e| e.as_str().into())
```

Comment:
I tried to make the caching more idiomatic for async, but that got very hairy very quickly... after many wasted hours, I gave up. Blocking in the foreground on a request should still only happen once for each cache entry when it's empty the first time.